tools/wake-format/emitter.cpp

```diff
@@ -337,16 +335,45 @@ wcl::doc Emitter::walk_token(ctx_t ctx, CSTElement node) {
 
 wcl::doc Emitter::walk_rhs(ctx_t ctx, CSTElement node) { return walk_placeholder(ctx, node); }
 
-wcl::doc Emitter::walk_apply(ctx_t ctx, CSTElement node) { return walk_placeholder(ctx, node); }
+wcl::doc Emitter::walk_apply(ctx_t ctx, CSTElement node) {
+  assert(node.id() == CST_APP);
+
+  return formatter()
+      .walk({CST_ID, CST_APP}, WALK(walk_node))
```

Comment:
You need an "is expression" predicate It can just be ID and APP for now but that's the set of nodes actully expected here, and you'll wind up needing that predicate over and over again. Make it a static function